src/client.js

```diff
@@ -124,6 +128,20 @@ let pushClient = {
 
     if (subscription) {
       await subscription.unsubscribe();
```

Comment:
If the user removes notifications I can imagine there would be no subscription to unsubscribe. If this is the case then the endpoint to unsubscribe will never get called when it could be.